src/compiler/utilities.ts

```diff
@@ -6444,6 +6444,10 @@ namespace ts {
         return !!(compilerOptions.declaration || compilerOptions.composite);
     }
 
+    export function getEmitPrettyDeclarations(compilerOptions: CompilerOptions): boolean {
+        return compilerOptions.prettyDeclaration !== undefined ? !!compilerOptions.prettyDeclaration : !!compilerOptions.declaration;

```

Comment:
Nope! `composite` declarations are only used for checking signatures, and not for human-readable output purposes; so I've explicitly not enabled the pretty pass when only that is set.

Reply:
when someone sets `composite` it also means emit `declaration`.. It isn't just for signature purpose

Reply:
That's a good point! Perhaps, we can use `getEmitDeclarations` directly instead of checking both `prettyDeclaration` and `declaration` separately:
```suggestion
        return getEmitDeclarations(compilerOptions);
```